modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c

```diff
@@ -25,47 +25,41 @@
 #define LOG_MODULE_NAME nrf_802154_temperature
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
-/** @brief Structure describing the temperature measurement module. */
-struct temperature {
-	const struct device *dev;	/**< Temperature device pointer. */
-	int8_t value;			/**< Last temperature measurement in degree Celsius. */
+static int8_t temperature_value = DEFAULT_TEMPERATURE;
 
-	struct k_work_delayable work;
-};
-
-static struct temperature temperature = {
-	.value = DEFAULT_TEMPERATURE,
-	.dev = DEVICE_DT_GET(DT_NODELABEL(temp))
-};
 
 #if defined(CONFIG_NRF_802154_TEMPERATURE_UPDATE)
+
+static const struct device *temperature_dev = DEVICE_DT_GET(DT_NODELABEL(temp));
+static struct k_work_delayable temperature_work;
```

Comment:
nit: it's pretty clear from the context that all data in this file exists for the purpose of temperature readouts, so IMO `temperature_` prefix is redundant. We can only leave `value`, `dev` and `work`.